bluesky/tests/test_devices.py

```diff
@@ -62,17 +62,47 @@ def test_separate_devices():
 
 
 @requires_ophyd
-def test_monitor():
+def test_monitor(fresh_RE):
     docs = []
     def collect(name, doc):
         docs.append(doc)
 
-    RE = setup_test_run_engine()
     a = A('')
     def plan():
         yield Msg('open_run')
         yield Msg('monitor', a.s1)
         a.s1._run_subs(sub_type='value')
         yield Msg('close_run')
-    RE(plan(), collect)
+    fresh_RE(plan(), collect)
     assert len(docs) == 4
+
+
+@requires_ophyd
+def test_monitor_with_pause_resume(fresh_RE):
+    docs = []
+    def collect(name, doc):
+        docs.append(doc)
+
+    a = A('')
+    def plan():
+        yield Msg('open_run')
+        yield Msg('monitor', a.s1)
+        yield Msg('checkpoint')
+        a.s1._run_subs(sub_type='value')
```

Comment:
Shouldn't this process synchronously?  I think this should work with only `null` message yielded instead of a sleep?